### DISCUSSION TITLE
HTTP Post Upload with UploadFile and alias

### DISCUSSION THREAD

**Discussion by nylocx** [fastapi/fastapi#11004] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
async def upload_docs(files: list[UploadFile] = File(..., alias="file")) -> dict:
    return await process_files(files)
```

### Description

I have used this snippet for multipart form data uploads and I am 90 % sure that it worked in the past. I could upload to this endpoint either with files or with file in the body. But currently I get:
`422 Unprocessable Entity`

and this response:
```json
{
  "detail": [
    {
      "type": "missing",
      "loc": [
        "body",
        "file"
      ],
      "msg": "Field required",
      "input": null,
      "url": "https://errors.pydantic.dev/2.5/v/missing"
    }
  ]
}
```

To test I used:

```
curl -X 'POST' \
  'http://localhost:8000/docs' \
  -H 'accept: application/json' \
  -H 'Content-Type: multipart/form-data' \
  -F 'file=@document.pdf;type=application/pdf'
```
and
```
curl -X 'POST' \
  'http://localhost:8000/docs' \
  -H 'accept: application/json' \
  -H 'Content-Type: multipart/form-data' \
  -F 'files=@document.pdf;type=application/pdf'
```

First one working second one failing.

Does anyone know if it should work like this or do I have something conceptually wrong here?

### Operating System

Linux, Windows

### FastAPI Version

0.105.0

### Pydantic Version

2.5.3

### Python Version

3.11.6

**Comment by Kludex** [comment, marked as answer]
- Follow the issue on https://github.com/tiangolo/fastapi/issues/10286.

**Reply by nylocx** [reply]
Thanks sorry for missing this.

**Reply by nylocx** [reply]
Should the workaround you posted be working on my example? I tried with
```python
async def upload_docs(
    files: list[UploadFile] = File(..., validation_alias="file")
) -> dict:
```
and
```python
async def upload_docs(
    files: list[UploadFile] = File(..., alias="file", validation_alias="file")
) -> dict:
```
without success.
